### PR TITLE
[Modules} Network Application-Gateway: Added private endpoint support

### PR DESCRIPTION
**Pull request by lsnoddy**
# Description

Added private endpoint support to the Network Application-Gateway module and added test in the common test case.
Part of the fix for #890.

### Pipeline references

| Pipeline |
| - |
| |
[Network - ApplicationGateways](https://github.com/lsnoddy/ResourceModules/actions/workflows/ms.network.applicationgateways.yml)

## Type of Change

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

File: modules/network/application-gateway/.test/common/main.test.bicep [hunk 1]
```diff
@@ -39,7 +42,7 @@ module nestedDependencies 'dependencies.bicep' = {
     publicIPName: 'dep-${namePrefix}-pip-${serviceShort}'
     managedIdentityName: 'dep-${namePrefix}-msi-${serviceShort}'
     certDeploymentScriptName: 'dep-${namePrefix}-ds-${serviceShort}'
-    keyVaultName: 'dep-${namePrefix}-kv-${serviceShort}'
+    keyVaultName: 'dep-${namePrefix}-kv-${serviceShort}-${substring(uniqueString(baseTime), 0, 3)}'
```
Comment (AlexanderSehr) [inline modules/network/application-gateway/.test/common/main.test.bicep:45]: Why the unique name? The far we only needed it in cases where purge protection was required

File: modules/network/application-gateway/main.bicep [hunk 2]
```diff
@@ -360,6 +365,29 @@ resource applicationGateway_diagnosticSettingName 'Microsoft.Insights/diagnostic
...
+  name: '${uniqueString(deployment().name, location)}-ApplicationGateway-PrivateEndpoint-${index}'
+  params: {
+    groupIds: [
+      privateEndpoint.service
+    ]
+    name: contains(privateEndpoint, 'name') ? privateEndpoint.name : 'pe-${last(split(applicationGateway.id, '/'))}-${privateEndpoint.service}-${index}'
+    serviceResourceId: applicationGateway.id
+    subnetResourceId: privateEndpoint.subnetResourceId
+    enableDefaultTelemetry: enableReferencedModulesTelemetry
+    location: reference(split(privateEndpoint.subnetResourceId, '/subnets/')[0], '2020-06-01', 'Full').location
```
Comment (ahmadabdalla) [inline modules/network/application-gateway/main.bicep:378]:
```suggestion
    location: contains(privateEndpoint, 'location') ? privateEndpoint.location : reference(split(privateEndpoint.subnetResourceId, '/subnets/')[0], '2020-06-01', 'Full').location
```

Comment (AlexanderSehr) [reply modules/network/application-gateway/main.bicep]: Good catch @ahmadabdalla

File: modules/network/application-gateway/.test/common/main.test.bicep (hunk 1 above)
Comment (lsnoddy) [reply modules/network/application-gateway/.test/common/main.test.bicep:45]: I ran into an error with the key vault being in a soft delete state, but must have been due to a manual deletion. I see that the removal pipeline process handles that. I'll go ahead and revert that change for the key vault name.

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**